test/integration/test_timeout.py

```diff
@@ -86,8 +86,9 @@ def get_authenticate_arguments(req):
         return req.mock_calls[0]
 
     @staticmethod
-    def get_request_arguments(tci, srv, idx):
-        tci.assertEqual(srv.http_client.request.call_count, idx + 1)
+    def get_request_arguments(tci, srv, idx, call_count=None):
```

Comment:
I think this function would benefit from a description of what `idx` and `call_count` actually mean.